airflow/models.py

```diff
@@ -726,6 +731,7 @@ def __init__(self, task, execution_date, state=None):
         self.pool = task.pool
         self.priority_weight = task.priority_weight_total
         self.try_number = 0
+        self.job_id = job_id
```

Comment:
It's important, the Job absctraction is actually the one looking for an external kill signal when clearing running tasks